crates/sqlexec/src/planner/context_builder.rs

```diff
@@ -70,6 +70,13 @@ impl<'a> PartialContextProvider<'a> {
                         }
                     }
                 }
+
+                if let Some(reader) = self.ctx.get_env_reader() {
+                    if let Some(table) = reader.resolve_table(table).unwrap() {
+                        return Ok(table);
+                    }
+                }
+
```

Comment:
how will this work if we add support for other dataframe libraries? 

it looks like this only allows for a single "env_reader"? I'd kinda expect that just to be a static struct, not a dyn trait. Maybe you had something else in mind though?

Reply:
oh nvm I think i answered my own question, the env reader is added to the sqlexec crate, so we could have a "python dataframe reader", or something that works for whatever other environment the exec is running in.